Makefile

```diff
@@ -169,7 +169,9 @@ fmt:
 
 # check-fmt: Checks gofmt/go fmt has been ran. gofmt -l lists files whose formatting differs from gofmt's, so it fails if there are unformatted go code.
 check-fmt:
-	@test -z $(shell gofmt -l ${GOFMT_DIRS})
+	@if [ $(shell gofmt -l ${GOFMT_DIRS} | wc -l ) != 0 ]; then \
+		echo "Error: there are unformatted go code., please run 'gofmt' before committing" ; \
```

Comment:
Minor typo (there shouldn't be a period after "go code").

Also, can you please squash your commits into one? We usually prefer one commit per PR. Thanks!